### PR TITLE
Add compat data for Element

### PR DESCRIPTION
**Pull request by Rambobafet**
Hello,

I updated the existing file `api/Element.json` for [Element](https://developer.mozilla.org/en-US/docs/Web/API/Element) created on [#1593](https://github.com/mdn/browser-compat-data/pull/1593).

Please note that's my first contribution for MDN, so I might have miss some good pratice. Let me know if that's the case :)

**Also I noticed that some method referenced on the data compat table of [Element](https://developer.mozilla.org/en-US/docs/Web/API/Element) doesn't exist in the file previously generated. Is it normal? Should I add the missing elements before merging? (`requestPointerLock` for example)**

### CODE REVIEW

**Comment by Rambobafet** [comment]
I don't understand this part from the linter : 
```
api/Element.json
  File : /home/travis/build/mdn/browser-compat-data/api/Element.json
  JSON schema – 2 error(s)
   item['api']['Element']['getElementsByTagName'] should NOT have additional properties
    item['api']['Element']['getElementsByTagNameNS'] should NOT have additional properties
```
Why is it not allowed? Did I miss a nesting level?
I "need" an additional property for [getElementsByTagName](https://developer.mozilla.org/fr/docs/Web/API/Element/getElementsByTagName)

File: api/Element.json [hunk 1]
```diff
@@ -1193,47 +1210,103 @@
             "standard_track": true,
             "deprecated": false
           }
+        },
+        "getElementsByTagName(*)": {
```
Comment (Elchi3) [inline api/Element.json]: Sorry about the unhelpful schema error messages. I think this is about identifiers which can't have characters like "(" or "*".

This should be something like:
 
```
"all_elements_selector": {
  "__compat": {
    "description": "getElementsByTagName(*)",
    "support": {}
  }
}
```

File: api/Element.json [hunk 2]
```diff
@@ -1244,47 +1317,76 @@
...
```
Comment (Elchi3) [inline api/Element.json]: Ditto.

**Comment by Rambobafet** [comment]
updated. :)

File: api/Element.json [hunk 3]
```diff
@@ -1193,47 +1210,104 @@
...
+              "standard_track": true,
+              "deprecated": false
+            }
+          }
         }
       },
       "getElementsByTagNameNS": {
         "__compat": {
-          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Element/getElementsByTagNameNS",
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Element/getElementsByTagName",
```
Comment (connorshea) [inline api/Element.json]: Wrong URL, keep the NS

**Comment by connorshea** [comment]
So this adds data for these subfeatures:
- `attributes`
- `className`
- `localName`
- `namespaceURI`
- `prefix`
- `slot`
- `tagName`
- `closest`
- `getAttribute`
- `getAttributeNames`
- `getAttributeNS`
- `getElementsByClassName`
- `getElementsByTagName`
- `getElementsByTagNameNS`
- `hasAttribute`
- `hasAttributes`
- `insertAdjacentText`

Also, it needs a rebase :)

**Comment by connorshea** [comment]
I've opened #2494 to track progress on the Element subfeatures.

**Comment by connorshea** [comment]
#2495 was merged, so this is going to have a decent amount of merge conflicts.

**Comment by Elchi3** [comment]
I've rebased this. Might be good if someone else checks if that went alright before I merge.

**Pull request merged**